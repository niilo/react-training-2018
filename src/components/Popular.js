import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {fetchPopularRepos} from '../util/api';
import RepoGrid from './RepoGrid';
import Loading from './Loading';

function SelectedLanguage(props) {
  const languages = [
    'All',
    'JavaScript',
    'Ruby',
    'Java',
    'CSS',
    'Python',
    'Golang',
  ];
  return (
    <ul className="languages">
      {languages.map(language => {
        return (
          <li
            key={language}
            style={
              language === props.selectedLanguage ? {color: '#d0021b'} : null
            }
            onClick={e => props.onSelect(language)}
          >
            {language}
          </li>
        );
      })}
    </ul>
  );
}

SelectedLanguage.propTypes = {
  selectedLanguage: PropTypes.string.isRequired,
  onSelect: PropTypes.func.isRequired,
};

class Popular extends Component {
  constructor(props) {
    super(props);
    if ('lang' in props) {
      this.state = {
        selectedLanguage: props.lang,
      };
    } else {
      this.state = {
        selectedLanguage: 'All',
        repos: null,
      };
    }
    //this.updateLanguage = this.updateLanguage.bind(this);
  }
  componentDidMount() {
    this.updateLanguage(this.state.selectedLanguage);
  }
  updateLanguage = lang => {
    fetchPopularRepos(lang).then(repos => {
      this.setState(() => {
        return {
          selectedLanguage: lang,
          repos,
        };
      });
    });
  };

  render() {
    return (
      <div>
        <SelectedLanguage
          selectedLanguage={this.state.selectedLanguage}
          onSelect={this.updateLanguage}
        />
        {!this.state.repos ? (
          <Loading text="downloading" speed={100} />
        ) : (
          <RepoGrid repos={this.state.repos} />
        )}
      </div>
    );
  }
}

export default Popular;
