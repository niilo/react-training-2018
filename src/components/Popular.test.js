import React from 'react';
import renderer from 'react-test-renderer';
import Popular from './Popular';

beforeEach(() => {
  window.fetch = jest.genMockFunction();
});

it('renders default correctly', () => {
  window.fetch.mockReturnValueOnce(Promise.resolve(new Response('{}')));
  const tree = renderer.create(<Popular />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders selected Javascriot correctly', () => {
  window.fetch.mockReturnValueOnce(Promise.resolve(new Response('{}')));
  const lang = 'JavaScript';
  const tree = renderer.create(<Popular lang={lang} />).toJSON();
  expect(tree).toMatchSnapshot();
});
