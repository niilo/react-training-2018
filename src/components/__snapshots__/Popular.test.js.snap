// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default correctly 1`] = `
<div>
  <ul
    className="languages"
  >
    <li
      onClick={[Function]}
      style={
        Object {
          "color": "#d0021b",
        }
      }
    >
      All
    </li>
    <li
      onClick={[Function]}
      style={null}
    >
      JavaScript
    </li>
    <li
      onClick={[Function]}
      style={null}
    >
      Ruby
    </li>
    <li
      onClick={[Function]}
      style={null}
    >
      Java
    </li>
    <li
      onClick={[Function]}
      style={null}
    >
      CSS
    </li>
    <li
      onClick={[Function]}
      style={null}
    >
      Python
    </li>
    <li
      onClick={[Function]}
      style={null}
    >
      Golang
    </li>
  </ul>
  <p
    style={
      Object {
        "fontSize": "35px",
        "textAlign": "center",
      }
    }
  >
    downloading
  </p>
</div>
`;

exports[`renders selected Javascriot correctly 1`] = `
<div>
  <ul
    className="languages"
  >
    <li
      onClick={[Function]}
      style={null}
    >
      All
    </li>
    <li
      onClick={[Function]}
      style={
        Object {
          "color": "#d0021b",
        }
      }
    >
      JavaScript
    </li>
    <li
      onClick={[Function]}
      style={null}
    >
      Ruby
    </li>
    <li
      onClick={[Function]}
      style={null}
    >
      Java
    </li>
    <li
      onClick={[Function]}
      style={null}
    >
      CSS
    </li>
    <li
      onClick={[Function]}
      style={null}
    >
      Python
    </li>
    <li
      onClick={[Function]}
      style={null}
    >
      Golang
    </li>
  </ul>
  <p
    style={
      Object {
        "fontSize": "35px",
        "textAlign": "center",
      }
    }
  >
    downloading
  </p>
</div>
`;
